### ISSUE TITLE
Flat namespace vs submodules

### ISSUE THREAD

**Issue by jorisvandenbossche** [Toblerity/Shapely#1225]
One item of the *"RFC: Roadmap for Shapely 2.0"* we didn't yet fully discuss is how to incorporate the vectorized functions of PyGEOS into Shapely (see also [this section](https://github.com/shapely/shapely-rfc/blob/df79554e19aa7182796eaa3fed22e0e7084607ef/rfc/0001-shapely-2.0-roadmap.md#incorporation-in-shapely)).

There are two aspects to this:

* where to put the vectorized functions? (currently in the top-level pygeos namespace)
* what to do with the existing scalar functions in shapely? (currently in submodules in shapely, eg `ops`, `algorithms`, `affinity`, `validation`, `wkt`/`wkb`, etc)

The obvious option for the first point is to follow PyGEOS and keep those new vectorized functions in the top-level namespace (as user-facing exposure). Although this breaks with the current pattern in Shapely (currently nothing is exposed in Shapely's top-level namespace, users need to import everything from some submodule), I would personally strongly argue for doing this. I think there are several reasons to do this: 1) user-friendly (less imports are needed; you don't need to know/remember for each function in which submodule it is located), 2) we don't need to create many new public submodules (currently many features are only available as method on the Geometry classes and not as function in a submodules, such as all predicates), 3) easier transition for people already using pygeos.

There are of course alternatives as mentioned in the RFC link (single submodule, multiple submodules as is the case right now, etc).

_If_ we go with the top-level namespace for the functions, the question is then what to do with the existing submodules in Shapely? (those can probably be discussed on a case by case depending on the outcome of the first question)

**Comment by sgillies** [comment]
Indeed, flatter namespaces are friendly to users. I support having the vectorized functions at the top level. 

The wkb/wkt modules: since they expose the same interface, it seems fine to have separate modules instead of wkt_dumps/wkb_dumps at the top level. In my experience, applications are using one serialization or the other, not a mix.

**Comment by jorisvandenbossche** [comment]
> Indeed, flatter namespaces are friendly to users. I support having the vectorized functions at the top level.

That sounds good!

> The wkb/wkt modules: since they expose the same interface, it seems fine to have separate modules instead of wkt_dumps/wkb_dumps at the top level. 

The top-level functions in pygeos are currently called `to_wk[b|t]` / `from_wk[b|t]`, and work on in-memory data (so are equivalent to `dumps` / `loads`, and not `dump`/`load`).

> what to do with the existing scalar functions in shapely? (currently in submodules in shapely, eg `ops`, `algorithms`, `affinity`, `validation`, `wkt`/`wkb`, etc)

Looking at a few of them:

* `ops` currently has those functions: 'linemerge', 'polygonize', 'polygonize_full', 'transform', 'unary_union', 'triangulate', 'voronoi_diagram', 'split', 'nearest_points', 'validate', 'snap', 'shared_paths', 'clip_by_rect', 'orient', 'substring'.
  * Most of those map to one of the vectorized functions from pygeos, and can probably eventually be deprecated in `ops` in favor of the top-level vectorized function (but that doesn't necessarily need to happen soon / in 2.0)
  * `split`, `substring` and `orient` are currently custom implemented operations that don't map directly to a GEOS function
* `algorithms.polylabel`: this is also a custom implementation, but this _might_ actually be similar as GEOS' new "MaximumInscribedCircle" (https://blog.cleverelephant.ca/2020/06/developer-life-2.html)
* `validity`: the tho functions here (`make_valid` and `explain_validity`) also map to a vectorized function
* `affinity`: this currently has some custom functionality to perform affine transformations (both general with custom matrix as a set of predefined ones). This might be worth keeping in a separate submodule?
